htdocs/user/passwordforgotten.php

```diff
@@ -131,6 +131,15 @@
 				$result = $edituser->fetch('', '', '', 1, $conf->entity, $username);
 			}
 
+			// If the user does not have the right to change his own password
+			if ($result > 0) {
+				$edituser->getrights('user');
+				if (!$edituser->hasRight('user', 'self', 'password')) {
+					$result = 0;
+					$edituser->error = 'USERNOTFOUND';
```

Comment:
May be we can use a different error code like 'USERNOTALLOWEDTOCHANGEPASS'  ?

Reply:
@eldy ready